src/utils/ApiClient.js

```diff
@@ -222,20 +226,37 @@ export default class ApiClient {
      * @private
      */
     processEncryptedResponse(httpMethod, err, res, callback) {
-        if (!err) {
-            let responseBody = res.rawResponse ? res.rawResponse : res.text;
-            try {
-                responseBody = this.encryption.base64Encode(JSON.parse(res.text));
-            } catch (e) {
-                // nothing to do
-            }
-            this.encryption.decrypt(responseBody)
-                .then((decryptedData) => {
-                    callback(undefined, JSON.parse(decryptedData.payload.toString()), decryptedData);
-                })
-                .catch(() => callback(`Failed to decrypt response for ${httpMethod} request`, responseBody, responseBody));
-        } else {
-            this.processNonEncryptedResponse(err, res, callback);
+        if (!res) {
+            callback("Try to decrypt empty response body", undefined, undefined);
         }
+        this.encryption.decrypt(res)
+            .then((decryptedData) => {
+                const responseBody = JSON.parse(decryptedData.payload.toString());
+                if (responseBody.errors) {
+                    const responseWithErrors = {};
+                    responseWithErrors.body = responseBody;
+                    this.processNonEncryptedResponse(responseBody, responseWithErrors, callback);
+                } else {
+                    callback(undefined, responseBody, decryptedData);
+                }
+            })
+            .catch(() => callback(`Failed to decrypt response for ${httpMethod} request`, res, res));
```

Comment:
why are we passing res parameter 2 times?

Reply:
For convension in callback we use 1st parameter for error, 2nd parameter for response body and 3rd  parameter for entire response. As this part of code is inside processEncryptedResponse method res parameter is actually the response body, so in that case we could pass 3rd parameter as undefined, but I suppose that in that case we can pass response body(res) as the response parameter